provider/apply.go

```diff
@@ -188,7 +188,13 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 		if ok {
 			err = s.waitForCompletion(ctx, wf, rs, rname, wt)
 			if err != nil {
-				return resp, err
+				resp.Diagnostics = append(resp.Diagnostics,
+					&tfprotov5.Diagnostic{
+						Severity: tfprotov5.DiagnosticSeverityError,
+						Detail:   err.Error(),
+						Summary:  fmt.Sprintf(`wait_for: timed out waiting on resource %q`, rnn),
```

Comment:
I wonder if there's some validation that can happen sooner, to prevent this error:

```
╷
│ Error: wait_for: timed out waiting on resource "default/tf-k8s-test-consul-server"

│ invalid field path "metadata.0.annotations.0.test": :1,10-11: Attribute name required; Dot must be followed by attribute
│ name.
╵
```

I used this as my config and it exited immediately:

```
  wait_for  = {
    timeout = "30s"
    fields  = {
      "metadata.0.annotations.0.test" = "true"
    }
  }
```

Also, is it possible for me to use an annotation to test this feature?

Reply:
Hmm, it looks like you need to use `[]` for indexes. We've documented it in the format you're using though so something must have changed.

Reply:
But yes – I'll add something to the ValidateResourceTypeConfig func to check for this.

Reply:
Have added validation so you get an error like this at plan:

```
╷
│ Error: Invalid "wait_for" configuration
│
│   with kubernetes_manifest.test,
│   on main.tf line 44, in resource "kubernetes_manifest" "test":
│   44:   wait_for = {
│   45:       fields = {
│   46:           "()*^B)*B^()^B)&*#^&$^#*&$TN&*#" = "lol"
│   47:       }
│   48:   }
│
│ invalid field path "()*^B)*B^()^B)&*#^&$^#*&$TN&*#": :1,4-5: Unsupported operator; Bitwise operators are not supported., and
│ 4 other diagnostic(s)
╵
```